src/metax_api/services/file_service.py

```diff
@@ -308,6 +308,32 @@ def destroy_bulk(cls, file_identifiers):
         _logger.info('Marked %d files as deleted from project %s' % (deleted_files_count, project_identifier))
         return Response({ 'deleted_files_count': deleted_files_count }, status=status.HTTP_200_OK)
 
+    @classmethod
+    def delete_project(cls, project_id):
+        """
+        Marks files deleted, deprecates related datasets and removes all directories.
+
+        This method is called by FileRPC
+        """
+        _logger.info('Begin to delete project from database...')
+
+        file_ids = [ id for id in File.objects.filter(project_identifier=project_id).values_list('id', flat=True) ]
+
+        deleted_files_count = 0
+
+        if file_ids:
+            deleted_files_count = cls._mark_files_as_deleted(file_ids)[0]
+            cls._mark_datasets_as_deprecated(file_ids)
```

Comment:
Perhaps overly cautious, but recommend executing `_mark_datasets_as_deprecated()` as the last action inside this if-block, since it publishes a message to rabbitmq. If anything fails in the following two method calls, the message has been published for no reason.